src/service/blockchain-event-listener-service.js

```diff
@@ -437,15 +484,95 @@ class BlockchainEventListenerService {
+                        };
+                    } catch (error) {
+                        this.logger.error(`Error parsing event data: ${error}`);
+                        return accumulator;
+                    }
+                }, {});
 
-                const { epochsNumber } = await this.blockchainModuleManager.getAgreementData(
-                    event.blockchainId,
+                const {
+                    blockchainId,
+                    assetContract: contract,
+                    tokenId,
+                    keyword,
+                    hashFunctionId,
+                    state: assertionId,
+                    startTime,
+                    epochsNumber,
+                    epochLength,
+                    // TODO: Uncomment when these arguments are added to the ServiceAgreementV1Created event
+                    // scoreFunctionId,
+                    // proofWindowOffsetPerc,
+                } = combinedData;
+
+                // TODO: Remove when added to the event
+                const scoreFunctionId = this.blockchainModuleManager.getScoreFunctionIds()[0];
+
+                // TODO: Remove when added to the event
+                const tx = await this.blockchainModuleManager.getTransaction(
+                    blockchainId,
+                    eventsGroup[0].transactionHash,
+                );
+                const assetCreator = tx.from;
+                const minProofWindowOffsetPerc =
+                    await this.blockchainModuleManager.getMinProofWindowOffsetPerc(blockchainId);
+                const maxProofWindowOffsetPerc =
+                    await this.blockchainModuleManager.getMaxProofWindowOffsetPerc(blockchainId);
+
+                const proofWindowOffsetPerc =
+                    minProofWindowOffsetPerc +
+                    (await this.blockchainModuleManager.generatePseudorandomUint8(
+                        blockchainId,
+                        assetCreator,
+                        tx.blockNumber,
+                        tx.timestamp,
+                        maxProofWindowOffsetPerc - minProofWindowOffsetPerc + 1,
+                    ));
+
+                const agreementId = this.serviceAgreementService.generateId(
+                    blockchainId,
+                    contract,
+                    tokenId,
+                    keyword,
+                    hashFunctionId,
+                );
+
+                await this.repositoryModuleManager.updateServiceAgreementRecord(
+                    blockchainId,
+                    contract,
+                    tokenId,
                     agreementId,
+                    startTime,
+                    epochsNumber,
+                    epochLength,
+                    scoreFunctionId,
+                    proofWindowOffsetPerc,
+                    hashFunctionId,
+                    keyword,
+                    assertionId,
+                    0,
+                    SERVICE_AGREEMENT_SOURCES.EVENT,
```

Comment:
Can we be in situation where we already have service agreement in database and we are overwriting data source with this?